api/controllers/developerController.js

```diff
@@ -0,0 +1,59 @@
+import ProjectDeveloperService from "../services/projectDeveloperService.js";
+
+class DeveloperController {
+  static async getAllDevelopers(req, res) {
+    try {
+      const developers = await ProjectDeveloperService.getAllDevelopers();
+      res.status(200).json(developers);
+    } catch (error) {
+      res.status(500).json({ message: error.message });
+    }
+  }
+
+  static async getClients(req, res) {
+    try {
+      const allClients = await ProjectDeveloperService.getAllClients();
+      res.status(200).json(allClients);
+    } catch (error) {
+      res.status(500).json({ message: error.message });
+    }
+  }
+
+  static async getDevelopersForProject(req, res) {
+    const { PJid: projectId } = req.params;
+    try {
+      const developers =
+        await ProjectDeveloperService.getDevelopersForProject(projectId);
+      res.status(200).json(developers);
+    } catch (error) {
+      res.status(500).json({ message: error.message });
+    }
+  }
+
+  static async assignDeveloper(req, res) {
+    const { project_id, developer_id } = req.body;
+    try {
+      const result = await ProjectDeveloperService.assignDeveloperToProject(
+        project_id,
+        developer_id
+      );
+      res.status(200).json(result);
+    } catch (error) {
+      res.status(500).json({ error: error.message });
+    }
+  }
+
+  static async removeDeveloper(req, res) {
+    const { PJid: projectId, DEVid: developerId } = req.params;
+    try {
+      const result = await ProjectDeveloperService.removeDeveloperFromProject(
+        projectId,
+        developerId
+      );
+      res.status(200).json(result);
+    } catch (error) {
+      res.status(500).json({ message: error.message });
```

Comment:
I'm unsure if when you would do `res.sendStatus(500)` is does not automatically send the error message 🤔 Might be worth checking

Reply:
i can check